aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/UploadTask.java

```diff
@@ -241,6 +242,8 @@ private Boolean uploadMultipartAndWaitForCompletion() throws ExecutionException
         } catch (final AmazonClientException ace) {
             LOGGER.error("Failed to complete multipart: " + upload.id
                     + " due to " + ace.getMessage(), ace);
+            abortMultiPartUpload(upload.id, upload.bucketName, upload.key,
```

Comment:
Make sure to catch any exceptions thrown by abort as well.

Reply:
The `abortMultiPartUpload` method catches all the exceptions thrown by `s3. abortMultiPartUpload`.